### PR TITLE
Very limited code fixup/cleanup on keylime_tenant CLI

### PR DESCRIPTION
**Pull request by maugustosilva**
This PR starts the cleanup and fixup of several small issues on `keylime_tenant` code. First of all, make sure all "jsonification" of resquest's responses happen at the static method `_jsonify_response` (renamed from `_print_json_response`) which can now optionally print the response. Second make sure this new function is invoked as often as possible throughout the CLI code, with obvious exceptions being documented. Third, the old `do_cv` method is renamed to `do_cvadd`, in order to improve the code's understandability and maintainability.

This by no means fixes all the problems and issues on `keylime_tenant`, but it is an incremental step in the right direction.

Signed-off-by: Marcio Silva <marcio.a.silva@ibm.com>

### CODE REVIEW

File: keylime/tenant.py [hunk 1]
```diff
@@ -589,6 +591,8 @@ def do_cvstatus(self):
 
         response = do_cvstatus.get((f"/v{self.api_version}/agents/{self.agent_uuid}"), timeout=self.request_timeout)
 
+        response_json = Tenant._jsonify_response(response)
+
```
Comment (stefanberger) [inline keylime/tenant.py:595]: there are two more `return response.json()` at lines 602 and 605 that probably should also be converted just like you converted line 613.

File: keylime/tenant.py [hunk 2]
```diff
@@ -640,6 +643,8 @@ def do_cvlist(self):
 
         response = do_cvstatus.get(f"/v{self.api_version}/agents/?verifier={verifier_id}", timeout=self.request_timeout)
 
+        Tenant._jsonify_response(response, True)
```
Comment (stefanberger) [inline keylime/tenant.py:646]: Why not `response_json = Tenant._jsonify_response(response)` here as well and then also convert lines 654 and 657, 665, 669,  etc.

File: keylime/tenant.py [hunk 3]
```diff
@@ -687,6 +692,9 @@ def do_cvbulkinfo(self):
             f"/v{self.api_version}/agents/?bulk={True}&verifier={verifier_id}", timeout=self.request_timeout
         )
 
+        # Bulkinfo can be quite large. Will not jsonify it without explicity need
+        # Tenant._jsonify_response(response, True)
```
Comment (stefanberger) [inline keylime/tenant.py:696]: lines 704, 707, 715, 717 all convert it to json.

For some odd reason the final return at line 734 doesn't seem to return a `response.json()` but a plain `response`. Maybe that case is never reached ? It's odd for sure if the function was to return  non-JSON there.

File: keylime/tenant.py [hunk 4]
```diff
@@ -915,9 +930,9 @@ def do_cvstop(self):
             logger.error("Verifier at %s with Port %s timed out.", self.verifier_ip, self.verifier_port)
             sys.exit()
 
-        response_body = response.json()
```
Comment (stefanberger) [inline keylime/tenant.py:918]: move line 920 `response_json = Tenant._jsonify_response(response, True)` to here...

File: keylime/tenant.py [hunk 5]
```diff
@@ -1162,26 +1177,28 @@ def do_add_allowlist(self, args):
...
         print(f"Show allowlist command response: {response.status_code}.")
-        Tenant._print_json_response(response)
+        Tenant._jsonify_response(response, True)
 
     @staticmethod
-    def _print_json_response(response):
+    def _jsonify_response(response, print_response=False):
         try:
             json_response = response.json()
         except ValueError:
```
Comment (stefanberger) [inline keylime/tenant.py:1197]: The difference to 'before' is that all `response.json()` you converted would have thrown a ValueError on bad input and now you change it to an empty map if there's an error . If you now raised the error in the except clause you would approximately get the same result again as before...

**Comment by stefanberger** [comment]
To get rid of the test failure you will probably need to rebase on master.

File: keylime/tenant.py (hunk 2 above)
Comment (maugustosilva) [reply keylime/tenant.py:646]: same, thank you

File: keylime/tenant.py (hunk 3 above)
Comment (maugustosilva) [reply keylime/tenant.py:696]: This function is in a somewhat fluid state. Right now, this is one of the very few function which returns (stdout) a (properly formatted) JSON string which can be easily consumed by a tool such as `jq`. I would like to make this the standard, but will reserve it for another PR.

File: keylime/tenant.py (hunk 5 above)
Comment (maugustosilva) [reply keylime/tenant.py]: Very good point. I am changing the behavior of the function to **optionally** raise an exception, thus keeping the usual behavior for the "other" methods. In the future, I will re-organize and standardize the behavior on all methods.

File: keylime/tenant.py [hunk 6]
```diff
@@ -743,36 +748,39 @@ def do_cvdelete(self, verifier_check=True):
         do_cvdelete = RequestsClient(self.verifier_base_url, True, tls_context=self.tls_context)
         response = do_cvdelete.delete(f"/v{self.api_version}/agents/{self.agent_uuid}", timeout=self.request_timeout)
 
-        response = response.json()
+        response_json = Tenant._jsonify_response(response, True)
```
Comment (stefanberger) [inline keylime/tenant.py:751]: You want this printed here?

File: keylime/tenant.py [hunk 7]
```diff
@@ -949,8 +961,7 @@ def do_quote(self):
                     do_quote = RequestsClient(cloudagent_base_url, tls_enabled=False)
                     response = do_quote.get(params, timeout=self.request_timeout)
 
-                print(response)
-                response_body = response.json()
+                response_json = Tenant._jsonify_response(response, False, True)
```
Comment (stefanberger) [inline keylime/tenant.py:964]: Here: no more printing ?

File: keylime/tenant.py [hunk 8]
```diff
@@ -743,36 +748,39 @@ def do_cvdelete(self, verifier_check=True):
...
-        if response["code"] == 202:
+        if response_json["code"] == 202:
             deleted = False
             for _ in range(12):
                 get_cvdelete = RequestsClient(self.verifier_base_url, True, tls_context=self.tls_context)
                 response = get_cvdelete.get(
                     f"/v{self.api_version}/agents/{self.agent_uuid}", timeout=self.request_timeout
                 )
 
+                response_json = Tenant._jsonify_response(response, False, True)
```
Comment (stefanberger) [inline keylime/tenant.py:771]: Here you convert it a 2nd time...

File: keylime/tenant.py (hunk 6 above)
Comment (stefanberger) [reply keylime/tenant.py:751]: This should probably throw an exception if conversion fails, so `(..., False, True)` ? A bit tricky this function now...

File: keylime/tenant.py (hunk 7 above)
Comment (maugustosilva) [reply keylime/tenant.py:964]: Correct, no more printing here

Comment (maugustosilva) [reply keylime/tenant.py:964]: Trying to keep it behaving as it was before

File: keylime/tenant.py (hunk 6 above)
Comment (maugustosilva) [reply keylime/tenant.py:751]: Yes, it should throw an exception, as it was before

File: keylime/tenant.py (hunk 8 above)
Comment (maugustosilva) [reply keylime/tenant.py:771]: It is a new response. This section of the code is a total mess, and I intend to fix that in the next PR :-)

Comment (stefanberger) [reply keylime/tenant.py:771]: ah, yes, right.

File: keylime/tenant.py (hunk 7 above)
Comment (stefanberger) [reply keylime/tenant.py:964]: was before printing: `print(response)`

Comment (maugustosilva) [reply keylime/tenant.py:964]: ah, very true... I was assuming this was a "leftover print", but you're right. Absent of clear evidence that it is indeed the case, we should retain the print out functionality

File: keylime/tenant.py [hunk 9]
```diff
@@ -569,12 +569,14 @@ def do_cv(self):
             logger.error("Verifier at %s with Port %s timed out.", self.verifier_ip, self.verifier_port)
             sys.exit()
 
+        response_json = Tenant._jsonify_response(response, False, True)
+
         if response.status_code == 409:
             # this is a conflict, need to update or delete it
-            Tenant._print_json_response(response)
+            json.dumps(response_json)
```
Comment (stefanberger) [inline keylime/tenant.py:576]: `print(json_response);` ?

File: keylime/tenant.py [hunk 10]
```diff
@@ -1162,26 +1174,31 @@ def do_add_allowlist(self, args):
         response = cv_client.post(
             f"/v{self.api_version}/allowlists/{self.ima_policy_name}", data=body, timeout=self.request_timeout
         )
-        Tenant._print_json_response(response)
+        Tenant._jsonify_response(response)
```
Comment (stefanberger) [inline keylime/tenant.py:1177]: This is kind of tricky to understand why we need it jsonify'd and it ends up printing it. Maybe don't print it in the function and do `print(Tenant._jsonify_response(response))` instead.

Comment (maugustosilva) [reply keylime/tenant.py:1177]: My longer term plan is to get agreement from others to have every command output a JSON-formatted string that is easily consumable by tools such as `jq` (think of `do_cvstatus` as the model). However, I am trying to contain the number of changes on this PR without disrupting the existing functionality. Please consider that at some point `_jsonify_response` will json.dumps (maybe even with indent, to make it more human readable).

File: keylime/tenant.py (hunk 9 above)
Comment (maugustosilva) [reply keylime/tenant.py:576]: Fair enough, keeping exactly the same functionality as before, I will change.

File: keylime/tenant.py (hunk 10 above)
Comment (maugustosilva) [reply keylime/tenant.py:1177]: Please note that I have an issue open with this purpose now https://github.com/keylime/keylime/issues/1159

**Review by galmasi — APPROVED** [review]
The renaming of functions is much appreciated.
Arguably there should be no print() calls in error handling code in keylime. However, it's a small sin; there are worse dragons in the code.

File: keylime/tenant.py (hunk 1 above)
Comment (aplanas) [reply keylime/tenant.py:595]: Same

File: keylime/tenant.py [hunk 11]
```diff
@@ -640,28 +643,29 @@ def do_cvlist(self):
 
         response = do_cvstatus.get(f"/v{self.api_version}/agents/?verifier={verifier_id}", timeout=self.request_timeout)
 
+        response_json = Tenant._jsonify_response(response, False, True)
```
Comment (aplanas) [inline keylime/tenant.py:646]: That is the 3rd call with reversed boolean parameters from the default.  Should be that a reversed defaults are better?

File: keylime/tenant.py [hunk 12]
```diff
@@ -569,12 +569,14 @@ def do_cv(self):
             logger.error("Verifier at %s with Port %s timed out.", self.verifier_ip, self.verifier_port)
             sys.exit()
 
+        response_json = Tenant._jsonify_response(response, False, True)
```
[the remark this answers is not in the document]
Comment (maugustosilva) [reply keylime/tenant.py:572]: Good point, will do

File: keylime/tenant.py (hunk 11 above)
Comment (maugustosilva) [reply keylime/tenant.py:646]: I elected to keep the "newest" calls, on the allowlist API (near the end of the file), as unmodified as possible.

File: keylime/tenant.py [hunk 13]
```diff
@@ -743,36 +748,39 @@ def do_cvdelete(self, verifier_check=True):
         do_cvdelete = RequestsClient(self.verifier_base_url, True, tls_context=self.tls_context)
         response = do_cvdelete.delete(f"/v{self.api_version}/agents/{self.agent_uuid}", timeout=self.request_timeout)
 
-        response = response.json()
+        response_json = Tenant._jsonify_response(response, False, True)
```
[the remark this answers is not in the document]
Comment (maugustosilva) [reply keylime/tenant.py:751]: At this point in time, the goal is to simply mimic the behavior of `.json()`

File: keylime/tenant.py [hunk 14]
```diff
@@ -1162,26 +1174,31 @@ def do_add_allowlist(self, args):
...
-        Tenant._print_json_response(response)
+        Tenant._jsonify_response(response)
 
     @staticmethod
-    def _print_json_response(response):
+    def _jsonify_response(response, print_response=True, raise_except=False):
         try:
             json_response = response.json()
-        except ValueError:
+        except ValueError as e:
```
[the remark this answers is not in the document]
Comment (maugustosilva) [reply keylime/tenant.py:1194]: Here is the crux of the matter: `tenant.py` is not in a good shape in my evaluation, and there will be more cleanup and fixup needed, including what will be required by https://github.com/keylime/keylime/issues/1158 and https://github.com/keylime/keylime/issues/1159 

I consider this to be a transitional and incremental PR, and that is why I have elected to not implement all that is needed at once. As it is, the PR lays the foundation for several of the changes, standardizes the treatment of request's response into a single static method, all without changing the current behavior.

**Review by mpeters — APPROVED** [review]

**Pull request merged**